server/backend/server.js

```diff
@@ -75,34 +73,33 @@ var findRoad = (parsed, db, callback) => {
     }
   };
 
-  segments.findOne(q, {speed: 1, road_id: 1, _id: 0}, (err, segment) => {
-    if (err) {
-      // Collection read error
-      console.log(err);
-      resp.online = 0;
-      callback(resp);
-    } else if (!segment) {
-      // No segment match!
-      callback(resp);
+  const resp = newResponse();
+  const segments = db.collection(config.segments);
+  const findOneSegment = denodeifyMethod(segments, 'findOne');
+
+  return findOneSegment(q, {speed: 1, road_id: 1, _id: 0})
+  .then({road_id, speed} => {
+    if (!segment) {
+      return resp;
     } else {
-      // Segment match!
-      var roads = db.collection(config.roads);
-
-      // Find road name using road_id
-      roads.findOne({_id: segment.road_id}, (err, road) => {
-        if (err || !road) {
-          // Read error, return what we have
-          console.log(err);
-          resp.name = "";
-        } else {
-          resp.found = 1;
-          resp.speed = segment.speed;
-          resp.name = road.name;
-        }
-
-        callback(resp);
-      });
+      const roads = db.collection(config.roads);
+      const findOneRoad = denodeifyMethod(roads, 'findOne');
+      return findOneRoad({_id: road_id})
```

Comment:
This is returning a `Promise`, however the promise already has some `thenables` chained onto the end. The return of last `thenable` or the return on the `catch` (if a `catch` is attached) is what gets returned to the consumer of this `findRoad`.